deployment-scanner/src/main/java/org/jboss/as/server/deployment/scanner/FileSystemDeploymentService.java

```diff
@@ -567,6 +576,21 @@ private boolean scan(boolean oneOffScan, final DeploymentOperations deploymentOp
         if (scanEnabled || oneOffScan) { // confirm the scan is still wanted
             ROOT_LOGGER.tracef("Scanning directory %s for deployment content changes", deploymentDir.getAbsolutePath());
 
+            // WFCORE-1579: skip the scan if deployment dir is not available
+            if (!checkDeploymentDir(deploymentDir)) {
+                if (lastScanSuccessful) {
+                    lastScanSuccessful = false;
+                    ROOT_LOGGER.scanFailed(deploymentDir.getAbsolutePath());
+                }
+                return scheduleRescan;
+            }
+            // if deployed content list was not established during scanner start (due to inaccessible deployment dir),
+            // do it now
+            if (!deployedContentEstablished) {
+                establishDeployedContentList(deploymentDir, deploymentOperations);
```

Comment:
@ehsavoie Can you review this PR please?

This bit here is the only concern I have. Not that it's necessarily wrong. Just that this is the one change where I think careful review is needed as it could introduce subtle bugs. And, if it's ok to do this here, then it seems unnecessary to do an establishDeployedContentList in startScanner() and L399-406 can just be dropped.